src/services/expenses.service.js

```diff
@@ -0,0 +1,69 @@
+const { data } = require('../data');
+
+const getAllExpenses = () => {
+  return data.expenses;
+};
+
+const getById = (id) => {
+  return data.expenses.find((exp) => exp.id === parseInt(id));
+};
+
+const normalizeCategory = (categories) => {
+  return Array.isArray(categories) || !categories ? categories : [categories];
+};
+
+const filterExpensesById = (id) => {
+  data.expenses = data.expenses.filter((exp) => exp.userId === parseInt(id));
```

Comment:
This function `filterExpensesById` is modifying the original data directly, which might lead to unexpected results. It's better to avoid direct modifications on original data. Consider returning a new array with the filtered results instead of modifying the original array.